src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -313,6 +319,31 @@ async Task<MsalAccessTokenCacheItem> ITokenCacheInternal.FindAccessTokenAsync(Au
             return null;
         }
 
+        private MsalAccessTokenCacheItem FilterATByKeyId(MsalAccessTokenCacheItem item, AuthenticationRequestParameters authenticationRequest)
+        {
+            if (item == null)
+            {
+                return null;
+            }
+
+            string requestKid = authenticationRequest.AuthenticationScheme.KeyId;
+            if (string.IsNullOrEmpty(item.Kid) && string.IsNullOrEmpty(requestKid))
+            {
+                authenticationRequest.RequestContext.Logger.Verbose("Bearer token found");
+                return item;
+            }
+
+            if (string.Equals(item.Kid, requestKid))
+            {
+                authenticationRequest.RequestContext.Logger.Verbose("PoP token found");
+                return item;
+            }
+
+            authenticationRequest.RequestContext.Logger.Info(
```

Comment:
is the $"" style string going to throw CultureInfo violations in the release build due to not using String.Format?